skt/reporter.py

```diff
@@ -434,61 +374,87 @@ def getjobresults(self):
-               'would like to help',
-               'test them. Below are the results of automatic tests we ran']
-        if self.mergedata['localpatch'] or self.mergedata['patchwork']:
-            msg[-1] += ' on a patchset'
-            msg += ['you\'re involved with, with hope it will help you find '
-                    'possible issues sooner.']
-        else:
-            # There is no patchset the person was involved with
-            msg[-1] += ', with hope it'
-            msg += ['will help you find possible issues sooner.']
+        j2_env = Environment(
+            loader=FileSystemLoader(SCRIPT_PATH),
+            trim_blocks=True,
+            lstrip_blocks=True
+        )
+        template = j2_env.get_template('templates/email_report.j2')
 
-        msg += ['\n'] + self.getmergeinfo()
+        # For a single report, we load up our existing config.
+        skt_states = [self.cfg]
 
-        if self.cfg.get("mergelog"):
-            msg += self.getmergefailure()
-        else:
-            self.get_kernel_config()
-            if self.cfg.get("buildlog"):
-                msg += self.getbuildfailure()
-            elif self.cfg.get('runner'):
-                msg += self.getjobresults()
-
-        msg += ['\nPlease reply to this email if you find an issue with our '
-                'testing process,',
-                'or wish to not receive these reports anymore.',
-                '\nSincerely,',
-                '  Kernel CI Team']
+        # For a multireport, we load up the configs provided by the user.
+        if self.multireport:
+            skt_states = [self.load_state_cfg(x) for x in self.statefiles]
+
+        reports = []
+        # Loop over each state and prepare the data for the report
+        for skt_state in skt_states:
+            # Load the state file and update our mergedata
+            self.cfg = skt_state
+            self.update_mergedata()
+
+            # Create a temporary report section that we can append at the end
+            temp_report = self.cfg
+            temp_report['mergedata'] = self.mergedata
+
+            # If the merge failed, get the merge log
+            temp_report['mergelog_output'] = None
+            if self.cfg.get('mergelog'):
+                with open(self.cfg.get('mergelog'), 'r') as fileh:
+                    temp_report['mergelog_output'] = fileh.read()
+
+            # Attach the kernel config to the report if the merge passed
+            if not self.cfg.get('mergelog'):
+                self.get_kernel_config()
+
+            # If the compile failed, get the build log
+            temp_report['buildlog_output'] = None
+            if self.cfg.get('buildlog'):
+                self.getbuildfailure()
+                with open(self.cfg.get('buildlog'), 'r') as fileh:
+                    temp_report['buildlog_output'] = fileh.read()
+
+            # Get the test results
+            temp_report['jobresults'] = self.getjobresults()
```

Comment:
This still causes failures for me :( In the original report, we are checking if the runner is specified before getting job results, but there's no check here so the report results in a traceback.